### PR TITLE
CI and Code Coverage workflow

### PR DESCRIPTION
**Pull request by btmonier**
Adds a GitHub action to:
* Build and unit test
* Generate a code coverage report and upload codecov.io

### CODE REVIEW

File: .github/workflows/phgv2_ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: PHGv2 CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types:
+      - opened
+    branches:
+      - main
```
Comment (zrm22) [inline .github/workflows/phgv2_ci.yml:9]: Should we add the path part we needed for the CD branch here as well so it only triggers to source code changes?  Might be good for now, but if we include docs might need to add more to it as we may want a trigger off documentation changes as well.

**Review by zrm22 — APPROVED** [review]

File: .github/workflows/phgv2_ci.yml (hunk 1 above)
Comment (btmonier) [reply .github/workflows/phgv2_ci.yml:9]: This has now been added

Comment (tcasstevens) [reply .github/workflows/phgv2_ci.yml:9]: Does this mean "opening a pull request against main"?

Comment (btmonier) [reply .github/workflows/phgv2_ci.yml:9]: @tcasstevens - I copied this from the BioKotlin repo. That is my interpretation as well

File: build.gradle.kts [hunk 2]
```diff
@@ -68,7 +68,7 @@ tasks.test {
 }
 
 tasks.withType<KotlinCompile> {
-    kotlinOptions.jvmTarget = "1.8"
+    kotlinOptions.jvmTarget = "11"
```
Comment (tcasstevens) [inline build.gradle.kts:71]: Are you sure about this? I don't know what it should be, but the Ci deployment is compiling with java-version 17

**Review by tcasstevens — APPROVED** [review]

File: build.gradle.kts (hunk 2 above)
Comment (btmonier) [reply build.gradle.kts:71]: @tcasstevens - should we update this to 17?

Comment (tcasstevens) [reply build.gradle.kts:71]: @btmonier If it compiles and works correctly I would say yes

**Pull request merged**